src/centralized.rs

```diff
@@ -0,0 +1,499 @@
+//! ALBA's bounded DFS scheme using Blake2b as hash function.
+//! (c.f. Section 3.2.2 of Alba paper)
+
+use rayon::prelude::*;
+use std::sync::atomic::{self, AtomicUsize};
+
+extern crate core;
+use crate::utils;
+
+use std::f64::consts::E;
+use std::sync::Arc;
+
+const DATA_LENGTH: usize = 32;
+const DIGEST_SIZE: usize = 32;
+
+type Data = [u8; DATA_LENGTH];
+type Hash = [u8; DIGEST_SIZE];
+
+/// Setup input parameters
+#[derive(Debug, Clone)]
+pub struct Params {
+    /// Soundness security parameter
+    pub lambda_sec: usize,
+    /// Completeness security parameter
+    pub lambda_rel: usize,
+    /// Approximate size of set Sp to lower bound
+    pub n_p: usize,
```

Comment:
Lambda technically is a number of bits so it is an integer.

Reply:
As for np, because it is used when indexing the bins, we get errors when changing to u64
```
error[E0277]: the type `[Vec<[u8; 32]>]` cannot be indexed by `u64`
   --> src/centralized.rs:299:24
    |
299 |         for &s in bins[round.h_usize].iter() {
    |                        ^^^^^^^^^^^^^ slice indices are of type `usize` or ranges of `usize`
    |
```

Reply:
Pushed changes for lambda and r

Reply:
I don't agree about lambda. The number of what bits? Suppose the prover needs to succeed only with probability 1/4. Then it is more efficient to set `lambda_rel` to log(4/3) instead of 1.

Reply:
If we make `n_p` `u64`, even though it will not let 32-bit provers generate a proof, 32-bit verifiers could still verify a certificate. It could in principle be useful. In the prover code we can just convert to `n_p` to `usize` for indexing into a slice. We don't even need to check that `n_p` fits in `usize` since if `set` really contains `n_p` elements, it must!

Reply:
In that case why not lambda as f32? Because of the precision? We should not need or want too large values for it

Reply:
Yes, it doesn't hurt to have more precision.